src/_common/content/content-set-cache.ts

```diff
@@ -38,3 +38,32 @@ export class ContentSetCache {
 		return this._length;
 	}
 }
+
+export class ContentSetCacheService {
+	private static _caches = new WeakMap<
+		ContentContainerModel,
+		Map<ContentContext, ContentSetCache>
+	>();
+
+	public static getCache(
```

Comment:
This entire thing is kind of unclear to me.
So if the cache doesnt exist on the first call, it creates the map and calls itself a second time. the second time it finds the map and initializes the cache and then calls itself a THIRD time to get the cache in the map?

That sounds a bit too complicated to follow for no good reason. Can we get rid of the recursion?

Reply:
Improved it :)